### PR TITLE
Wrap up initial data access refactor

### PR DESCRIPTION
**Pull request by jon-hagg**
### Purpose

Relates to https://github.com/Breakthrough-Energy/PowerSimData/issues/552

Based on the changes introduced in the `data_access_refactor` feature branch, this contains bugfixes and some code cleanup. Additionally it introduces blob storage as a read only source of scenario data, using fs-azureblob to maintain consistency across filesystems. 

### What the code is doing
* Fix bugs encountered while launching simulations in a container and in preparing one on the server. 
* Remove the `move_to` method which is only being used for the scenario/execute list, and combine with `push` which is what is used for those files
* use `MultiFS` to combine remote filesystems (represented by the `DataAccess.fs` attribute)
* make code coverage report opt in
* new unit test for creating/preparing a scenario using an ephemeral filesystem for local and "remote" storage. only external dependency is the `test_usa_tamu` profiles from blob storage
* enable switch to delete state once scenario is created, and unit test for deletion

### Testing
Ran a simulation in a container, and created/prepared one on the server. For the container, I removed the `pip install .` from the Dockerfile and mounted the PowerSimData repo on my local machine into `/PowerSimData` in the client container, so changes could be tested more quickly. However this is optional and shouldn't affect functionality. 

### Time estimate
40 min

### CODE REVIEW

File: powersimdata/data_access/data_access.py [hunk 1]
```diff
@@ -177,14 +158,14 @@ def checksum(self, relative_path):
         :param str relative_path: path relative to root
         :return: (*str*) -- the checksum of the file
         """
-        return "dummy_value"
+        return self.fs.hash(relative_path, "sha256")
```
Comment (danielolsen) [inline powersimdata/data_access/data_access.py:161]: Did we really have `"dummy_value"` running in prod?

Comment (jon-hagg) [reply powersimdata/data_access/data_access.py:161]: Yeah.. although we override by using `sha1sum` if using the server. The dummy value was only used for local/container environments.

**Comment by kasparm** [comment]
When preparing the scenario from my machine I get:
```
>>> scenario.create_scenario()
CREATING SCENARIO: test | dummy1

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/kmueller/EGM/PowerSimData/powersimdata/scenario/create.py", line 116, in create_scenario
    self._scenario_list_manager.add_entry(self._scenario_info)
  File "/Users/kmueller/EGM/PowerSimData/powersimdata/data_access/csv_store.py", line 18, in wrapper
    checksum = self.data_access.checksum(self._FILE_NAME)
  File "/Users/kmueller/EGM/PowerSimData/powersimdata/data_access/data_access.py", line 289, in checksum
    return self.fs.checksum(full_path)
AttributeError: 'MultiFS' object has no attribute 'checksum'
```

**Comment by kasparm** [comment]
Now:
```
>>> scenario.create_scenario()
CREATING SCENARIO: test | dummy1

Transferring ScenarioList.csv from ssh_fs
100%|#####################################################################################| 669k/669k [00:00<00:00, 3.19Mb/s]
--> Adding entry in ScenarioList.csv
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/kmueller/EGM/PowerSimData/powersimdata/scenario/create.py", line 116, in create_scenario
    self._scenario_list_manager.add_entry(self._scenario_info)
  File "/Users/kmueller/EGM/PowerSimData/powersimdata/data_access/csv_store.py", line 20, in wrapper
    self.commit(table, checksum)
  File "/Users/kmueller/EGM/PowerSimData/powersimdata/data_access/csv_store.py", line 76, in commit
    self.data_access.push(tmp_name, checksum, rename=self._FILE_NAME)
  File "/Users/kmueller/EGM/PowerSimData/powersimdata/data_access/data_access.py", line 294, in push
    self._check_file_exists(backup, should_exist=False, mode="w")
  File "/Users/kmueller/EGM/PowerSimData/powersimdata/data_access/data_access.py", line 183, in _check_file_exists
    raise OSError(f"{path} already exists on {location}")
OSError: ScenarioList.csv.temp already exists on ssh_fs
```

**Comment by jon-hagg** [comment]
Couple things:
1) the `.temp` file should be deleted when we update the scenario list. not sure why that didn't happen so I did it manually.
2) based on the stack trace, i think you are on a previous version of the branch before I force pushed (it wasn't up very long, thought I would get away with that). don't think it would cause this particular issue, but still recommend doing a pull

**Comment by kasparm** [comment]
I had done a pull but maybe somehow the `.temp` got left behind.
Now I see the `exec_command` is  missing for `MultiFS`.
Full stack trace:
```
>>> scenario.create_scenario()
CREATING SCENARIO: test | dummy2

Transferring ScenarioList.csv from ssh_fs
100%|#####################################################################################| 669k/669k [00:00<00:00, 3.04Mb/s]
--> Adding entry in ScenarioList.csv
Transferring ScenarioList.csv to server
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/kmueller/EGM/PowerSimData/powersimdata/scenario/create.py", line 116, in create_scenario
    self._scenario_list_manager.add_entry(self._scenario_info)
  File "/Users/kmueller/EGM/PowerSimData/powersimdata/data_access/csv_store.py", line 20, in wrapper
    self.commit(table, checksum)
  File "/Users/kmueller/EGM/PowerSimData/powersimdata/data_access/csv_store.py", line 76, in commit
    self.data_access.push(tmp_name, checksum, rename=self._FILE_NAME)
  File "/Users/kmueller/EGM/PowerSimData/powersimdata/data_access/data_access.py", line 313, in push
    _, _, stderr = self.fs.exec_command(command)
AttributeError: 'MultiFS' object has no attribute 'exec_command'
```

**Review by kasparm — COMMENTED** [review]
When preparing scenario in client server mode I can successfully prepare it but after a while I get the socket exception:
```
>>> scenario.prepare_simulation_input()
Transferring ExecuteList.csv from ssh_fs
100%|####################################################################################| 50.5k/50.5k [00:00<00:00, 692kb/s]
---------------------------
PREPARING SIMULATION INPUTS
---------------------------
--> Loading demand
Writing scaled demand profile to tmp/scenario_3911/demand.csv
Writing tmp/scenario_3911/demand.csv
--> Loading hydro
Writing scaled hydro profile to tmp/scenario_3911/hydro.csv
Writing tmp/scenario_3911/hydro.csv
--> Loading solar
Writing scaled solar profile to tmp/scenario_3911/solar.csv
Writing tmp/scenario_3911/solar.csv
--> Loading wind
Writing scaled wind profile to tmp/scenario_3911/wind.csv
Writing tmp/scenario_3911/wind.csv
Building MPC file
Writing tmp/scenario_3911/case.mat
Transferring ExecuteList.csv from ssh_fs
100%|####################################################################################| 50.5k/50.5k [00:00<00:00, 717kb/s]
--> Setting status=prepared in execute list
Transferring ExecuteList.csv to server
>>> Socket exception: Operation timed out (60)
```
and every consecutive call such as `scenario.print_scenario_info()` returns `OSError: Socket is closed`

It looks like the scenario was prepared properly but I still get the operation timed out error.

File: powersimdata/scenario/execute.py [hunk 2]
```diff
@@ -193,42 +194,37 @@ def __init__(self, data_access, scenario_info, grid, ct):
...
             tp = TransformProfile(self._scenario_info, self.grid, self.ct)
             profile = tp.get_profile(kind)
             print(f"Writing scaled {kind} profile to {filepath}")
-            self._data_access.write(filepath, profile, save_local=False)
+            with self._data_access.write(filepath, save_local=False) as f:
+                profile.to_csv(f)
         else:
             from_dir = self._data_access.tmp_folder(profile_as)
-            src = self._data_access.join(from_dir, f"{kind}.csv")
+            src = "/".join([from_dir, file_name])
```
Comment (dmuldrew) [inline powersimdata/scenario/execute.py:229]: Will this introduce an issue with Window backslash paths?

**Comment by jon-hagg** [comment]
Deleting a scenario is still extremely slow. I will look through pyfilesystem source to see how it's iterating glob matches.

**Comment by kasparm** [comment]
Deleting is much faster now. Thank you!

**Review by kasparm — APPROVED** [review]
Looks good.
Thank you for your work!

**Pull request merged**